pkg/pillar/types/global.go

```diff
@@ -103,312 +107,417 @@ func (gs *GlobalStatus) UpdateItemValuesFromGlobalConfig(gc GlobalConfig) {
+}
+
+func GetDefaultSpecs() ConfigItemSpecMap {
+	var configItemSpecMap ConfigItemSpecMap
+	configItemSpecMap.AddIntItem("timer.config.interval", "", 60, 5, 0xFFFFFFFF)
+	configItemSpecMap.AddIntItem("timer.metric.interval", "", 60, 5, 0xFFFFFFFF)
+	configItemSpecMap.AddIntItem("timer.send.timeout", "", 0, 0, 0xFFFFFFFF)
+	configItemSpecMap.AddIntItem("timer.reboot.no.network", "", 7*24*3600, 120, 0xFFFFFFFF)
+	configItemSpecMap.AddIntItem("timer.update.fallback.no.network", "", 300, 5, 0xFFFFFFFF)
+	configItemSpecMap.AddIntItem("timer.test.baseimage.update", "", 60, 5, 0xFFFFFFFF)
+	configItemSpecMap.AddIntItem("timer.port.georedo", "", 3600, 60, 0xFFFFFFFF)
+	configItemSpecMap.AddIntItem("timer.port.georetry", "", 600, 5, 0xFFFFFFFF)
+	configItemSpecMap.AddIntItem("timer.port.testduration", "", 30, 10, 0xFFFFFFFF)
+	configItemSpecMap.AddIntItem("timer.port.testinterval", "", 300, 300, 0xFFFFFFFF)
+	configItemSpecMap.AddIntItem("timer.port.timeout", "", 15, 60, 0xFFFFFFFF)
+	configItemSpecMap.AddIntItem("timer.port.testbetterinterval", "", 0, 0, 0xFFFFFFFF)
+	configItemSpecMap.AddTriStateItem("network.fallback.any.eth", "", TS_ENABLED)
+	configItemSpecMap.AddTriStateItem("network.allow.wwan.app.download", "", TS_ENABLED)
+	configItemSpecMap.AddTriStateItem("network.allow.wwan.baseos.download", "", TS_ENABLED)
+	configItemSpecMap.AddBoolItem("debug.enable.usb", "", true)
+	configItemSpecMap.AddBoolItem("debug.enable.usb", "", true)
+	configItemSpecMap.AddBoolItem("app.allow.vnc", "", false)
+	configItemSpecMap.AddIntItem("app.allow.vnc", "", 60, 5, 0xFFFFFFFF)               //UNSURE OF VALUES
+	configItemSpecMap.AddIntItem("timer.use.config.checkpoint", "", 60, 5, 0xFFFFFFFF) //UNSURE OF VALUES
+	configItemSpecMap.AddIntItem("timer.gc.download", "", 600, 60, 0xFFFFFFFF)
+	configItemSpecMap.AddIntItem("timer.gc.vdisk", "", 3600, 60, 0xFFFFFFFF)
+	configItemSpecMap.AddIntItem("timer.gc.rkt.graceperiod", "", 3600, 600, 0xFFFFFFFF)
+	configItemSpecMap.AddIntItem("timer.download.retry", "", 600, 0, 0xFFFFFFFF)
+	configItemSpecMap.AddIntItem("timer.boot.retry", "", 600, 10, 0xFFFFFFFF)
+	configItemSpecMap.AddIntItem("network.allow.wwan.app.download", "", 0, 0, 0xFFFFFFFF)    //UNSURE OF VALUES
+	configItemSpecMap.AddIntItem("network.allow.wwan.baseos.download", "", 0, 0, 0xFFFFFFFF) //UNSURE OF VALUES
+	configItemSpecMap.AddStringItem("debug.default.loglevel", "loglevel", "info")
+	configItemSpecMap.AddStringItem("debug.default.remote.loglevel", "remote.loglevel", "info")
+	configItemSpecMap.AddIntItem("storage.dom0.disk.minusage.percent", "", 20, 20, 0xFFFFFFFF)
+	configItemSpecMap.AddIntItem("storage.apps.ignore.disk.check", "", 0, 5, 80)
+	return configItemSpecMap
+}
+
+func GetDefaultValueMap(configMap ConfigItemSpecMap) *ConfigItemValueMap {
+	var valueMap ConfigItemValueMap
+	var item ConfigItemValue
+	for key, value := range configMap.GlobalSettings {
+		ParseSpecToValue(value)
+		valueMap.GlobalSettings[key] = item
```

Comment:
This should really be:
     valueMap.GlobalSettings[key] = configSpecItem.DevaultItemValue()